measures/ResStockArgumentsPostHPXML/resources/hvac_flexibility/setpoint_modifier.rb

```diff
@@ -47,27 +47,49 @@ def modify_setpoints(setpoints, flexibility_inputs)
         if heating_setpoint[index] > cooling_setpoint[index]
           cooling_setpoint[index] = heating_setpoint[index]
         end
+        heating_setpoint[index] = _clip_setpoints(day_type, heating_setpoint[index])
       else
         cooling_setpoint[index] += _get_setpoint_offset(index, 'cooling', offset_times, flexibility_inputs)
         # If the offset causes the set points to be inverted, adjust the heating setpoint to correct it
         # This can happen during pre-cooling if originally the cooling and heating setpoints were the same
         if heating_setpoint[index] > cooling_setpoint[index]
           heating_setpoint[index] = cooling_setpoint[index]
         end
+        cooling_setpoint[index] = _clip_setpoints(day_type, cooling_setpoint[index])
       end
     end
     { heating_setpoint: heating_setpoint, cooling_setpoint: cooling_setpoint }
   end
 
+  def _clip_setpoints(day_type, setpoint)
+    if day_type == 'heating'
+      if setpoint > 80
+        setpoint = 80
+      elsif setpoint < 55
+        setpoint = 55
+      end
+      return setpoint
+    end
+    if day_type == 'cooling'
+      if setpoint > 80
+        setpoint = 80
+      elsif setpoint < 60
+        setpoint = 60
+      end
+      return setpoint
+    end
```

Comment:
I think this can be simplified a bit to

```suggestion
return 80 if day_type == 'heating' && setpoint > 80
return 55 if day_type == 'heating' && setpoint < 55
return 80 if day_type == 'cooling' && setpoint > 80
return 60 if day_type == 'cooling' && setpoint < 60
setpoint
```

I think it's cleaner to have a flat else-if and early return when possible.

Reply:
sounds good!